### PR TITLE
chore: Revert #174

### PR DESCRIPTION
**Pull request by grabbou**
Reverts react-native-community/react-native-cli#174

Unfortunately, it's causing some issues, reverting to debug further.

### CODE REVIEW

File: packages/cli/src/link/unlink.js [hunk 1]
```diff
@@ -87,13 +85,13 @@ function unlink(args: Array<string>, ctx: ContextT) {
...
     platforms = getPlatforms(ctx.root);
   } catch (err) {
-    logger.error(
+    log.error(
       "No package.json found. Are you sure it's a React Native project?"
     );
     return Promise.reject(err);
   }
 
-  const allDependencies = getProjectDependencies(ctx.root).map(dependency =>
```
Comment (thymikee) [inline packages/cli/src/link/unlink.js]: @grabbou please revert this change, it brings back the unlink bug

**Comment by grabbou** [comment]
Shipped in 1.2.4. If you are affected by this, just reinstall your `node_modules`.
